### PR TITLE
Move a lot of Reader to Redux

### PR DESCRIPTION
**Pull request by NickHeiner**
Big changes moving to use Redux more in core parts of the app. We wanted to do this in a simpler way, but the existing app had enough connections that once you started to move part over, it wasn't practical to do anything but push the rest through.

Nothing else in Reader should be merged until this is in, so we should accept a few minor issues / tech improvement tickets will come out of this. We can merge this, then circle back immediately and fix those as well.

Connect #1432 

This will make #1469 and #1470 easier as well.

I won't be here on Friday (4/28) but other people should feel free to push to this branch and merge it.

### CODE REVIEW

File: client/app/components/Pdf.jsx [hunk 1]
```diff
@@ -207,8 +207,6 @@ export class Pdf extends React.Component {
           this.props.onPageChange(1, pdfDocument.pdfInfo.numPages);
         }
 
-        this.props.handleSetCurrentRenderedFile(this.props.documentId);
```
Comment (mdbenjam) [inline client/app/components/Pdf.jsx]: Why are we removing this? This sets the currentRenderedFile, which I think is still necessary in this file. It serves a different purpose, than the rest of the documents in the store. This is used to make sure we're rendering the pages from the correct document.

File: client/app/reader/DecisionReviewer.jsx [hunk 2]
```diff
@@ -55,55 +55,11 @@ export class DecisionReviewer extends React.Component {
...
-    const currentPdfIndex = Math.min(this.state.currentPdfIndex + 1,
-        this.state.documents.length - 1);
-
-    this.setPage(currentPdfIndex);
-  }
-
-  // This method is used for updating attributes of documents.
-  // Since we maintain a sorted and unsorted list of documents
-  // when we update one, we need to update the other.
-  setDocumentAttribute = (pdfNumber, field, value) => {
```
Comment (mdbenjam) [inline client/app/reader/DecisionReviewer.jsx]: Thank god this is gone.

File: client/app/reader/reducer.js [hunk 3]
```diff
@@ -23,12 +23,41 @@ export const initialState = {
   documents: {}
 };
 
+const updateFilteredDocIds = (nextState) => {
+  const activeCategoryFilters = _(nextState.ui.pdfList.filters.category).
+        toPairs().
+        filter((([key, value]) => value)). // eslint-disable-line no-unused-vars
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Try [,value] to get rid of the eslint error. http://eslint.org/docs/rules/no-unused-vars

File: client/app/reader/reducer.js [hunk 4]
```diff
@@ -38,7 +67,48 @@ export default (state = initialState, action = {}) => {
...
+          sortBy: {
+            $set: action.payload.sortBy
+          },
+          sortAscending: {
+            $apply: (prevVal) => !prevVal
+          }
+        }
+      }
+    }));
+  case Constants.SELECT_CURRENT_VIEWER_PDF:
```
Comment (NickHeiner) [inline client/app/reader/reducer.js]: ✅

File: client/app/reader/PdfListView.jsx [hunk 5]
```diff
@@ -308,6 +308,14 @@ const mapStateToProps = (state) => ({
 });
 
 const mapDispatchToProps = (dispatch) => ({
+  changeSortState(sortBy) {
```
Comment (NickHeiner) [inline client/app/reader/PdfListView.jsx]: ✅

File: client/app/components/Pdf.jsx (hunk 1 above)
Comment (NickHeiner) [inline client/app/components/Pdf.jsx]: ✅

**Comment by NickHeiner** [comment]
Right now this is failing with:

```
File /home/travis/build/department-of-veterans-affairs/caseflow/app/controllers/reader/documents_controller.rb is only 86% covered.                   This is below the expected minimum coverage per file of 90%
```

I believe it is because the change I made to the spec tests means the following code is no longer being run:

```rb
  def single_document
    Document.find(params[:id]).tap do |t|
      t.filename = params[:filename]
      t.type = params[:type]
      t.received_at = params[:received_at]
    end
  end
  helper_method :single_document
```

@mdbenjam, we decided to degrade/drop this functionality for now until we could get `react-router` going. I'll let you decide how to fix this situation.

File: client/app/reader/reducer.js [hunk 6]
```diff
@@ -28,17 +91,92 @@ export default (state = initialState, action = {}) => {
...
+        }
+      }
+    });
+  case Constants.SET_PDF_READY_TO_SHOW:
+    return update(state, {
+      ui: {
+        pdf: {
+          pdfsReadyToShow: {
+            [action.payload.docId]: {
+              $set: true
```
Comment (mdbenjam) [inline client/app/reader/reducer.js]: Actually, I think this is fine. We set this to false above when we switch PDFs.

**Pull request merged**